### PR TITLE
Remove static nodes, deprecate --nodes

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1228504

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by fabianofranz** [comment]
LGTM

**Comment by liggitt** [comment]
test flaked in https://github.com/openshift/origin/issues/6339
[merge]

File: contrib/vagrant/provision-master.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ os::provision::init-certs "${CONFIG_ROOT}" "${NETWORK_PLUGIN}" \
 echo "Launching openshift daemons"
 NODE_LIST=$(os::provision::join , ${NODE_NAMES[@]})
 cmd="/usr/bin/openshift start master --loglevel=${LOG_LEVEL} \
- --master=https://${MASTER_IP}:8443 --nodes=${NODE_LIST} \
```
Comment (marun) [inline contrib/vagrant/provision-master.sh]: This change renders line 32 unnecessary.

Comment (liggitt) [inline contrib/vagrant/provision-master.sh]: true. follow-ups welcome :)

**Pull request merged**